File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Armstrong Number 1/ARMSTRONG_1.py
import math
while 1 != 0:
    try:
        n = int(input())
        temp = n
        cnt = 0
        Sum = 0
        
        while temp != 0:
            cnt += 1
            temp //= 10
        temp = n
        while temp != 0:
            Sum += pow(temp % 10, cnt)
            temp //= 10
        if int(Sum) == n:
            print("YES")
        else:
            print("NO")
    except EOFError:
        break

File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Armstrong Number 1/ARMSTRONG_1.cs
using System;

namespace Armstrong_1
{
    class Program
    {
        static void Main(string[] args)
        {
            int n;
            while (int.TryParse(Console.ReadLine(), out n))
            {
                int temp = n;
                int cnt = 0;
                
                while (temp != 0)
                {
                    cnt++;
                    temp /= 10;
                }
                
                temp = n;
                int sum = 0;
                
                while (temp != 0)
                {
                    sum += (int)Math.Pow(temp % 10, cnt);
                    temp /= 10;
                }

                if (sum == n)
                    Console.WriteLine("YES");
                else
                    Console.WriteLine("NO");
            }
        }
    }
}
